core-web/libs/sdk/client/README.md

```diff
@@ -169,237 +233,280 @@ const { pageAsset, content } = await client.page.get('/about-us', {
 
-```javascript
-// Fetch navigation using the page.get method with GraphQL
-const { content } = await client.page.get('/', {
-    languageId: '1',
-    graphql: {
-        content: {
-            nav: `
-                query {
-                    nav {
-                        identifier
-                        path
-                        label
-                        children {
-                            identifier
-                            path
-                            label
-                        }
-                    }
-                }
-            `
-        }
-    }
-});
+The `getCollection` method uses immutable patterns, so each method call returns a new instance. This means you can reuse partial queries by storing them before adding additional filters.
 
-// Access navigation data
-console.log(content.nav);
-```
+**Builder Methods:**
 
-> **Note:** The legacy `client.nav.get()` method is still available but deprecated.
+| Method       | Arguments                | Description                        |
+| ------------ | ------------------------ | ---------------------------------- |
+| `query()`    | `string` \| `BuildQuery` | Filter content using query builder |
+| `limit()`    | `number`                 | Set page size                      |
+| `page()`     | `number`                 | Set page number                    |
+| `sortBy()`   | `SortBy[]`               | Order results                      |
+| `language()` | `number \| string`       | Set content language               |
 
-#### Legacy Navigation Example
+**Usage:**
 
-```javascript
-// Legacy approach - still works but is deprecated
-const navData = await client.nav.get({
-    path: '/',
-    depth: 2,
-    languageId: 1
-});
+```typescript
+const blogs = await client.content
+    .getCollection('Blog')
+    .query((qb) =>
+        qb.field('title').contains('dotCMS').and().field('publishDate').greaterThan('2023-01-01')
+    )
+    .limit(10)
+    .page(1)
+    .fetch();
 ```
 
-### Fetching a Collection of Content
-
-The `getCollection` method allows you to fetch a collection of content items (sometimes called "contentlets") using a builder pattern for complex queries.
-
-#### Basic Usage
-
-Here's a simple example to fetch content from a collection:
+**With TypeScript:**
 
 ```typescript
-import { dotCMSCreateClient } from '@dotcms/client';
+import { DotCMSBasicContentlet } from '@dotcms/types';
+
+// Define your content type interface
+interface BlogPost extends DotCMSBasicContentlet {
+    title: string;
+    publishDate: string;
+    author: string;
+    blogContent: {
+        json: string;
+    };
+    urlTitle: string;
+    tags: string[];
+}
 
-const client = dotCMSCreateClient({
-    dotcmsUrl: 'https://your-dotcms-instance.com',
-    authToken: 'your-auth-token'
+// Use the generic type parameter for type safety
+const response = await client.content.getCollection<BlogPost>('Blog').fetch();
```

Comment:
`fetch` is not a valid method, the correct one is `then`. If you use `await`, you don't need to call `then`